### DISCUSSION TITLE
reset "the first onBlur" event of " mode: onTouch" after calling reset()

### DISCUSSION THREAD

**Discussion by 6thpath** [react-hook-form/react-hook-form#3458] [Ideas] [answered]
Hi Bill, thanks for the excellent library. I really love it.

I'm not sure if this's a bug or not and I decided to create a discussion. I'm curious about the reset function. This's my current configs: 
```js
const { reset } = useForm({
  mode: 'onTouch',
  shouldFocusError: true,
})
```

after a successful submission, I called `reset()` (I've tried both case, with args and none arg). I think that the `reset` should work like a re-init function, I mean it restart the `onTouch` mode, trigger validation on the first `onBlur` and attach validation `onChange` after.

 My situation is I'd like to render an icon as suffix of my input, this is the logic 
```js
errors[fieldName] ? <IconX /> : formState.touched[fieldName] ? <IconCheck> :  undefined
```
this input is registered with a `required` validation, everything work properly until I have a successful submission and call `reset()`,
after that I `focus` on the input and `blur`, the validation not triggered because it only validate `onChange` but I need it to validate on the first `blur` event an `onChange` later on like before the `reset`.

**Comment by bluebill1049** [comment]
hey buddy, any chance you can share a codesandbox for what you try to achieve?

**Comment by 6thpath** [comment]
Hi, thanks for the quick reply.
Here's the [Codesandbox](https://codesandbox.io/s/reproduce-react-hook-form-ufzu0?file=/src/App.js)
Steps to reproduce:
- `focus` on the input and `blur` => You'll see the `required` message show up (touched = true, error = true)
- `focus` again and type something then click `Submit` => `reset()` get called
- `focus` and `blur` like step 1 => only touched added but no validation (touched = true, error = false)

**Comment by bluebill1049** [comment, marked as answer]
hey, take a look at the codesandbox, it's correct to call reset at useEffect: https://codesandbox.io/s/reproduce-react-hook-form-forked-tqs9u?file=/src/App.js

there is a CSB with submit reset: https://react-hook-form.com/api#reset

**Reply by 6thpath** [reply]
Oh I see it, I didn't read the documentations thoroughly and missed that point, Sorry for taking your time, have a good day.

P/s: The library is amazing, I've stuck in many issues while using uncontrolled input until I found your library. Keep up the good work buddy xD

**Reply by bluebill1049** [reply]
no worries at all mate 👍 feel free to ask question.
